src/node/index.ts

```diff
@@ -8,65 +8,51 @@
 /* eslint-disable filenames/match-exported */
 
 import { Buffer } from "buffer";
-import * as fs from "fs";
+import * as fs from "fs/promises";
 
 import Bag from "../Bag";
 import BagReader from "../BagReader";
-import { Callback } from "../types";
+import { Filelike } from "../types";
 
 // reader using nodejs fs api
-export class Reader {
+export class Reader implements Filelike {
   _filename: string;
-  _fd: number | null | undefined;
+  _file?: fs.FileHandle;
```

Comment:
With this change, jest now dumps a bunch of these:

```
(node:42051) [DEP0137] DeprecationWarning: Closing a FileHandle object on garbage collection is deprecated. Please close FileHandle objects explicitly using FileHandle.prototype.close(). In the future, an error will be thrown if a file descriptor is closed during garbage collection.
(node:42051) Warning: Closing file descriptor 25 on garbage collection
(node:42051) Warning: Closing file descriptor 28 on garbage collection
...
```

Should we care about these? I suspect we were already leaking file handles with the numeric `fd`s, but now we are just getting an explicit warning?

Currently the Bag API doesn't provide a close().

Reply:
How are we suppose to close the file handle object then? There are no explicit destructors in javascript...

Reply:
We could provide a close method on bag and ensure that each test manually closes a bag when it is done. We'd also have to closing even after something throws an error in the error tests...